ui/hooks/snaps/useSnapNavigation.ts

```diff
@@ -1,18 +1,32 @@
 import { useHistory } from 'react-router-dom';
 import { parseMetaMaskUrl } from '@metamask/snaps-utils';
+import browser from 'webextension-polyfill';
 import { getSnapRoute } from '../../helpers/utils/util';
+import {
+  ENVIRONMENT_TYPE_NOTIFICATION,
+  EXTENSION_MESSAGES,
+} from '../../../shared/constants/app';
 
 const useSnapNavigation = () => {
   const history = useHistory();
-  const navigate = (url: string) => {
+  const navigate = (url: string, envType: string, onCancel?: () => void) => {
     let path;
     const linkData = parseMetaMaskUrl(url);
     if (linkData.snapId) {
       path = getSnapRoute(linkData.snapId);
     } else {
       path = linkData.path;
     }
-    history.push(path);
+    if (envType === ENVIRONMENT_TYPE_NOTIFICATION) {
+      browser.runtime.sendMessage({
```

Comment:
This is done to avoid making the open popup call within the same execution context of the dialog (notification) window, otherwise chrome will throw an error. It has to be made from the background.